pkg/profiling/config/config.go

```diff
@@ -0,0 +1,69 @@
+package config
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Configuration struct {
+	RESTConfig *rest.Config
+	Client     client.Client
+	Scheme     *runtime.Scheme
+}
+
+func (c *Configuration) Load() error {
+	// creates the in-cluster config
+	restConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return err
+	}
```

Comment:
Any hesitance towards always relying on the in-cluster config? The only immediate concern would be attempting to develop locally and running the binary from a remote host, which should error/produce a panic if it cannot load the in-cluster configuration. The latter can be implemented as a follow-up later down the line, but trying to wrap my head around whether there are similar situations to look out for here.

Reply:
It seems reasonable if people decide to use this binary outside of containers, but I'd like to see this work tacked on post code freeze.